packages/base-driver/lib/wd-proxy/proxy.js

```diff
@@ -0,0 +1,404 @@
+import _ from 'lodash';
+import axios from 'axios';
+import {logger, util} from '@appium/support';
+import {
+  errors,
+  isErrorType,
+  errorFromW3CJsonCode,
+  getResponseForW3CError,
+} from '../protocol/errors';
+import {routeToCommandName} from '../protocol';
+import {MAX_LOG_BODY_LENGTH, DEFAULT_BASE_PATH} from '../constants';
+import http from 'http';
+import https from 'https';
+
+const DEFAULT_LOG = logger.getLogger('WD Proxy');
+const DEFAULT_REQUEST_TIMEOUT = 240000;
+const COMPACT_ERROR_PATTERNS = [/\bECONNREFUSED\b/, /socket hang up/];
+
+const ALLOWED_OPTS = [
+  'scheme',
+  'server',
+  'port',
+  'base',
+  'reqBasePath',
+  'sessionId',
+  'timeout',
+  'log',
+  'keepAlive',
+];
+
+export class WDProxy {
```

Comment:
We could potentially move the common stuff into a base class and then inherit both WDProxy and JWProxy from it to avoid unnecessary code duplication

Reply:
i'm sort of hoping we'll just delete jwproxy at some point, which would then entail undoing the abstraction

Reply:
I would only consider doing that as soon as all internal drivers switch to full w3c support. The current situation is that we have no one, for example, to maintain/update appium remote debugger (iOS Safari stuff). Also, I observe some customers enforcing chromedriver to work in jwp mode.